packages/shared/lib/contexts/onboarding/actions/restoreBackupFromStrongholdFile.ts

```diff
@@ -1,19 +1,34 @@
 import { get } from 'svelte/store'
 
+import { getStorageDirectoryOfProfile } from '@core/profile'
 import { profileManager, restoreBackup } from '@core/profile-manager'
 
 import { onboardingProfile, updateOnboardingProfile } from '../stores'
-import { validateStrongholdCoinType } from '../helpers'
+import { restoreBackupByCopyingFile, validateStrongholdCoinType } from '../helpers'
 
-export async function restoreBackupFromStrongholdFile(strongholdPassword: string): Promise<void> {
-    const { importFilePath, networkProtocol } = get(onboardingProfile)
-    await restoreBackup(importFilePath, strongholdPassword)
+export async function restoreBackupFromStrongholdFile(
+    strongholdPassword: string,
+    byCopyingFile = false
+): Promise<void> {
+    const { id, importFilePath, networkProtocol, clientOptions } = get(onboardingProfile)
+    if (byCopyingFile) {
+        const storageDirectory = await getStorageDirectoryOfProfile(id)
+        await restoreBackupByCopyingFile(
+            importFilePath,
+            storageDirectory,
+            strongholdPassword,
+            clientOptions,
+            profileManager
+        )
+    } else {
+        await restoreBackup(importFilePath, strongholdPassword)
 
-    /**
-     * NOTE: We must check that the Stronghold file's coin_type matches the current networkProtocol
-     * and is not based on an alternate protocol.
-     */
-    await validateStrongholdCoinType(get(profileManager), networkProtocol)
+        /**
+         * NOTE: We must check that the Stronghold file's coin_type matches the current networkProtocol
+         * and is not based on an alternate protocol.
+         */
+        await validateStrongholdCoinType(get(profileManager), networkProtocol)
 
-    updateOnboardingProfile({ lastStrongholdBackupTime: new Date() })
+        updateOnboardingProfile({ lastStrongholdBackupTime: new Date() })
```

Comment:
Shouldn't this be outside the else statement?

Reply:
It's tricky because only the `else` branch code actually "restores" from the backup whereas the `if` branch copies the file but we still have to recover everything since it was wasn't technically restored. If we set this, this it'll show to users like they don't need to create a backup when they likely do if they're using the copy-file method.